nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/dynamodb/PutDynamoDBRecord.java

```diff
@@ -0,0 +1,365 @@
+    static final AllowableValue SORT_BY_SEQUENCE = new AllowableValue("by-sequence", "Generate sequence",
+            "The processor will assign a number for every item based on the original record's position in the incoming FlowFile. The field name is determined by \"Sort Key Field\" as sort key.");
+
+    static final PropertyDescriptor RECORD_READER = new PropertyDescriptor.Builder()
+            .name("record-reader")
+            .displayName("Record Reader")
+            .description("Specifies the Controller Service to use for parsing incoming data and determining the data's schema.")
+            .identifiesControllerService(RecordReaderFactory.class)
+            .required(true)
+            .build();
+
+    static final PropertyDescriptor PARTITION_STRATEGY = new PropertyDescriptor.Builder()
+            .name("partition-strategy")
+            .displayName("Partition Strategy")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .allowableValues(PARTITION_BY_FIELD, PARTITION_BY_ATTRIBUTE, PARTITION_GENERATED)
+            .defaultValue(PARTITION_BY_FIELD.getValue())
+            .description("Defines the strategy the processor uses to assign partition key to the inserted Items. Partition key is also known as hash key.")
+            .build();
+
+    static final PropertyDescriptor PARTITION_FIELD = new PropertyDescriptor.Builder()
+            .name("partition-field")
+            .displayName("Partition Field")
+            .required(true)
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .description("The name of the Item field will be used as partition key. Depending on the \"Partition Strategy\" this might be a field from the incoming Record or a generated one.")
+            .build();
+
+    static final PropertyDescriptor PARTITION_ATTRIBUTE = new PropertyDescriptor.Builder()
+            .name("partition-attribute")
+            .displayName("Partition Attribute")
+            .required(true)
+            .dependsOn(PARTITION_STRATEGY, PARTITION_BY_ATTRIBUTE)
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .description("Specifies the FlowFile attribute will be used as the value of the partition key when using \"Partition by attribute\" partition strategy.")
+            .build();
+
+    static final PropertyDescriptor SORT_KEY_STRATEGY = new PropertyDescriptor.Builder()
+            .name("sort-key-strategy")
+            .displayName("Sort Key Strategy")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .allowableValues(SORT_NONE, SORT_BY_FIELD, SORT_BY_SEQUENCE)
```

Comment:
For picking strategy, neither of them supports attributes as usual in case of restriced value set (a list of allowable values). For other related properties (sort key field, etc.) it should support and as I see it does.